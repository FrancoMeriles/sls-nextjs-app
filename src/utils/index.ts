import currency from 'currency.js'

const getRandomColor = () => 'hsl(' + Math.round(Math.random() * 359) + ',100%,50%)'

const getPesoFormatted = (value) => currency(value, { separator: '.', decimal: ',', symbol: '$ ' })

const getPriceFormatted = (price) => (price ? getPesoFormatted(price).format() : 0)

const checkIfProductExistInFavorite = (products, productId) =>
  products.some((product) => product.id === productId)

const getPriceRound = (highPrice, lowPrice) =>
  Math.round(((highPrice - lowPrice) / highPrice) * 100)

const getStarsToRate = (rate) => {
  const QUANTITY_STARS = 5
  return [...Array(QUANTITY_STARS).keys()].map((star) =>
    rate >= star + 1 ? 'full' : rate > star ? 'half' : 'empty'
  )
}

export {
  getRandomColor,
  getPriceFormatted,
  getPriceRound,
  getStarsToRate,
  checkIfProductExistInFavorite,
}
