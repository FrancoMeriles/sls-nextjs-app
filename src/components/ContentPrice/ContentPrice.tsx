import { FC } from 'react'
import { getPriceFormatted, getPriceRound } from '@base/utils'
import { Box, Text, Stat, StatNumber, Tag } from '@chakra-ui/react'

interface ContentPriceProps {
  original_price: number
  price: number
}

const ContentPrice: FC<ContentPriceProps> = ({ original_price, price }) => {
  return (
    <>
      <Text as="del" fontSize="sm" color="gray.500" mt={3} d="block">
        {original_price && <p>{getPriceFormatted(original_price)}</p>}
      </Text>
      <Stat lineHeight="9">
        <Box display="flex">
          <StatNumber fontSize="3xl">{getPriceFormatted(price)}</StatNumber>
          {original_price && (
            <Tag size="sm" bg="transparent" color="brand.200">
              {getPriceRound(original_price, price)} %OFF
            </Tag>
          )}
        </Box>
      </Stat>
    </>
  )
}

export default ContentPrice
