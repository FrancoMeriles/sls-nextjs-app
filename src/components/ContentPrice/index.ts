import ContentPrice from './ContentPrice'

export default ContentPrice
