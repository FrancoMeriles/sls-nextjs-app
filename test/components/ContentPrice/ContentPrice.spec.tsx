import { render, screen } from '@testing-library/react'
import { getPriceFormatted, getDiscountedPrice } from '@base/utils'

import ContentPrice from '@base/components/ContentPrice'
import renderer from 'react-test-renderer'

const mockContentPriceProps = {
  original_price: 1000,
  price: 900,
}

describe('ContentPrice', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<ContentPrice {...mockContentPriceProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data if the price has discount', async () => {
    render(<ContentPrice {...mockContentPriceProps} />)
    expect(
      screen.getByText(getPriceFormatted(mockContentPriceProps.original_price))
    ).toBeInTheDocument()
    expect(screen.getByText(getPriceFormatted(mockContentPriceProps.price))).toBeInTheDocument()
    expect(
      screen.getByText(
        `${getDiscountedPrice(
          mockContentPriceProps.original_price,
          mockContentPriceProps.price
        )} %OFF`
      )
    ).toBeInTheDocument()
  })

  it('should renders and display the correct data if the price doesnt have discount', async () => {
    mockContentPriceProps.original_price = 0
    render(<ContentPrice {...mockContentPriceProps} />)
    expect(screen.getByText(getPriceFormatted(mockContentPriceProps.price))).toBeInTheDocument()
    expect(screen.queryByText(`%OFF`)).not.toBeInTheDocument()
  })
})
