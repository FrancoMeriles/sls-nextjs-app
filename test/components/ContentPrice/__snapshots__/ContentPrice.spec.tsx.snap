// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPrice should render and match snapshot with required props 1`] = `
Array [
  <del
    className="chakra-text css-1njsmg6"
  >
    <p>
      $ 1.000,00
    </p>
  </del>,
  <div
    className="chakra-stat css-1pdpmw6"
  >
    <dl>
      <div
        className="css-k008qs"
      >
        <dd
          className="chakra-stat__number css-wp3asv"
        >
          $ 900,00
        </dd>
        <span
          className="css-118lf1r"
        >
          10
           %OFF
        </span>
      </div>
    </dl>
  </div>,
]
`;
